Client/qtTeamTalk/preferencesdlg.cpp

```diff
@@ -489,6 +489,19 @@ void PreferencesDlg::initDisplayTab()
     }
     QString lang = ttSettings->value(SETTINGS_DISPLAY_LANGUAGE, SETTINGS_DISPLAY_LANGUAGE_DEFAULT).toString();
     int index = ui.languageBox->findData(lang);
+    if (index < 0)
+    {
+        QString langPrefix = lang.section('_', 0, 0);

```

Comment:
What is the purpose of this code? Why is the language split using "_"?

Reply:
E.G. when I start TT on my system with a blank config, SETTINGS_DISPLAY_LANGUAGE is set to fr_FR, so if I open and than close preferences, language in reset to default because fr_FR is not corresponding to an item in the combobox, so we should check for a partial item

Reply:
Doesn't this cause problems with pt_BR and pt_PT? Which one does it choose?

Reply:
Partial check is only done if no item is fully equal to settings, so if you put pt_BR in the config file "portuguès" will be selected in the combo, and for pt_PT it will be "Portuguès europeu". Same for Chinese with zh_CN and zh_TW

Reply:
Ok, can't you use `ui.languageBox->findData(langPrefix);` instead of the for-loop?

Reply:
Done